scripts/check-title-labels.lean

```diff
@@ -0,0 +1,106 @@
+  if !known_kinds.contains (parts.get! 0) then
+    IO.println s!"error: the PR kind should be one of {", ".intercalate (known_kinds.map (s!"\"{·}\""))}"
+    numberErrors := numberErrors + 1
+  return numberErrors
+
+
+def has_contradictory_labels(labels : List String) : Bool := Id.run do
+  -- Combine common labels.
+  let substitutions := [
+    ("ready-to-merge", "bors"), ("auto-merge-after-CI", "bors"),
+    ("blocked-by-other-PR", "blocked"), ("blocked-by-core-PR", "blocked"),
+    ("blocked-by-batt-PR", "blocked"), ("blocked-by-qq-PR", "blocked"),
+  ]
+  let mut canonicalise : Lean.HashMap String String := .ofList substitutions
+  let normalised_labels := labels.map fun label ↦ (canonicalise.findD label label)
+  -- Test for contradictory label combinations.
+  if normalised_labels.contains "awaiting-review-DONT-USE" then
+    return true
+  -- Waiting for a decision contradicts most other labels.
+  else if normalised_labels.contains "awaiting-zulip" && ["awaiting-author", "delegated", "bors", "WIP"].any normalised_labels.contains then
+    return true
+  else if normalised_labels.contains "WIP" && ["awaiting-review", "bors"].any normalised_labels.contains then
+    return true
+  else if ["awaiting-author", "awaiting-zulip"].all normalised_labels.contains then
+    return true
+  else if ["bors", "WIP"].all normalised_labels.contains then
+    return true
+  else
+    return false
+
+
+open Cli in
+/-- Implementation of the `mk_all` command line program.
+The exit code is the number of files that the command updates/creates. -/
+def checkTitleLabelCLI (args : Parsed) : IO UInt32 := do
+  let title := (args.positionalArg! "title").value
+  let labels : String := (args.positionalArg! "labels").value
+  let labels := labels.splitOn " "
+  let mut numberErrors := 0
+  if !labels.contains "WIP" then
+    -- We do not complain about WIP PRs.
+    -- The title should be of the form "abbrev: main title" or "abbrev(scope): main title".
+    numberErrors := ← validate_title title
+    -- A feature PR should have a topic label.
+    if title.startsWith "feat" && !labels.any (fun s ↦ s == "CI" || s.startsWith "t-") then
+      IO.println "error: feature PRs should have a 't-something' or the 'CI' label"
+      numberErrors := numberErrors + 1
+  -- Check for contradictory labels.
+  if has_contradictory_labels labels then
+    IO.println "error: PR labels are contradictory"
+    numberErrors := numberErrors + 1
+  return numberErrors
+
+open Cli in
+/-- Setting up command line options and help text for `lake exe check-title-label`. -/
+def checkTitleLabel : Cmd := `[Cli|
+  «check-title-label» VIA checkTitleLabelCLI; ["0.0.1"]
+  "Check that a PR title matches the formatting requirements.
+  If this PR is a feature PR, also verify that it has a topic label."
+
+  ARGS:
+    title : String; "this PR's title"
+    labels : String; "space-separated list of label names of this PR"
```

Comment:
```suggestion
    ...labels : String; "Label name(s) of this PR"
```
I think this should give you an `Array String` of labels which you can access with

```lean
let labels : Array String := args.variableArgsAs! String
```

so you don't need to split manually.

See `outputs` in [import-graph](https://github.com/leanprover-community/import-graph/blob/0a756018b736ef4f4bc3c9a45fc03cf97a0b1a0c/Main.lean#L28-L33) for an example.

Reply:
Ah, that's the syntax for passing an array of arguments. Much nicer indeed, thank you!